File: apps/mobile/types/watched-process.ts
export interface WatchedProcess {
  id: string;
  correlationKey: string;
  displayName: string;
  processType: string;
  watchedSince: string;       // ISO-8601
  notifyOnUpdate: boolean;
  lastViewedAt?: string;      // ISO-8601 — null until user opens the tracker
}

File: apps/services/PoliTickIt.Domain/Models/WatchedProcess.cs
namespace PoliTickIt.Domain.Models;

public class WatchedProcess
{
    /// <summary>Deterministic composite: first 16 hex chars of SHA256("{userId}:{correlationKey}").</summary>
    public string Id { get; set; } = string.Empty;
    public string UserId { get; set; } = string.Empty;
    public string CorrelationKey { get; set; } = string.Empty;
    public string DisplayName { get; set; } = string.Empty;
    public string ProcessType { get; set; } = string.Empty;
    public DateTime WatchedSince { get; set; } = DateTime.UtcNow;
    public bool NotifyOnUpdate { get; set; } = true;
    public DateTime? LastViewedAt { get; set; }
}
